src/rust/ide/view/graph-editor/src/builtin/visualization/java_script/histogram.js

```diff
@@ -316,6 +317,18 @@ class Histogram extends Visualization {
                         transformedScale.y = distanceScale.rescaleY(transformedScale.y)
                     }
                 }
+            } else if (
+                d3.event.sourceEvent != null &&
+                d3.event.sourceEvent.buttons === midButtonClicked
+            ) {
+                const distanceScale = d3.zoomIdentity.translate(
+                    d3.event.sourceEvent.movementX / 2,
+                    d3.event.sourceEvent.movementY / 2
+                )
+                transformedScale.x = distanceScale.rescaleX(transformedScale.x)
+                if (distanceScale.rescaleY(transformedScale.y).domain()[0] >= 0) {
+                    transformedScale.y = distanceScale.rescaleY(transformedScale.y)
+                }
```

Comment:
This function became very repetitive. And I don't understand all lines, for example: why we need to divide by two? was that just picked for the best UX, or is there other reason? And why we have `if` for x rescale, but for y not?

Reply:
We have if for y rescale because we dont want the user to get lower than 0 on the histogram - it cant show values below 0, so that's why. 2 is because this way the mouse stays exactly above the panned element, and the velocity is correct.